api/spitsgids/SpitsgidsController.php

```diff
@@ -0,0 +1,135 @@
+    }
+
+    private static function processFeedbackOneConnection($feedback) {
+        self::feedbackOneConnectionToOccupancyTable($feedback);
+        self::feedbackOneConnectionToFeedbackTable($feedback);
+    }
+
+    private static function feedbackOneConnectionToOccupancyTable($feedback) {
+        $m = new MongoDB\Driver\Manager("mongodb://localhost:27017");
+        $occupancy = new MongoDB\Collection($m, 'spitsgids', 'occupancy');
+
+        $id = $feedback["vehicle"] . "-" . $feedback["date"] . "-" . basename($feedback["from"]);
+        $occupancyExists = $occupancy->findOne(array('id' => $id));
+
+        $occupancyData = array(
+            'id' => $id,
+            'vehicle' => $feedback["vehicle"],
+            'from' => $feedback["from"],
+            'date' => $feedback["date"],
+            'feedback' => self::occupancyToNumber($feedback["occupancy"]),
+            'feedbackAmount' => 1,
+            'occupancy' => self::occupancyToNumber($feedback["occupancy"])
+        );
+
+        if(is_null($occupancyExists)) {
+            $occupancy->insertOne($occupancyData);
+        } else {
+            if(is_null($occupancyExists["feedback"])) {
+                $occupancy->updateOne(
+                    array('id' => $id),
+                    array('$set' => array('feedback' => $occupancyData["feedback"], 'occupancy' => $occupancyData["feedback"]))
+                );
+            } else {
+                $feedbackAmount = $occupancyExists->feedbackAmount + 1;
+                $feedbackScore = ((double)$occupancyExists->feedback * (double)$occupancyExists->feedbackAmount + (double)$occupancyData["feedback"]) / (double)$feedbackAmount;
+
+                $occupancy->updateOne(
+                    array('id' => $id),
+                    array('$set' => array('feedbackAmount' => $feedbackAmount, 'feedback' => $feedbackScore, 'occupancy' => $feedbackScore))
+                );
+            }
+        }
+    }
+
+    private static function feedbackOneConnectionToFeedbackTable($feedback) {
+        $m = new MongoDB\Driver\Manager("mongodb://localhost:27017");
+        $feedbackTable = new MongoDB\Collection($m, 'spitsgids', 'feedback');
+
+        $id = $feedback["vehicle"] . "-" . $feedback["date"] . "-" . basename($feedback["from"]);
+
+        $feedbackData = array(
+            'id' => $id,
+            'vehicle' => $feedback["vehicle"],
+            'from' => $feedback["from"],
+            'date' => $feedback["date"],
+            'occupancy' => self::occupancyToNumber($feedback["occupancy"])
+        );
+
+        $feedbackTable->insertOne($feedbackData);
+    }
+
+    private static function occupancyToNumber($occupancy) {
+        switch ($occupancy) {
+            case 'https://api.irail.be/terms/low':
+                return 0;
+                break;
+            case 'https://api.irail.be/terms/medium':
+                return 1;
+                break;
+            default:
+                return 2;
```

Comment:
They can only send low medium or high and at this point it is already to late to throw error messages because this code is executed after a 201 response has been sent. What is your proposal to improve this code?